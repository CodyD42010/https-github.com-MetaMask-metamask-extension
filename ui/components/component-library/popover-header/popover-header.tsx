import React from 'react';
import classnames from 'classnames';
import { HeaderBase, Text, ButtonIcon, ButtonIconSize, IconName } from '..';
import {
  TextVariant,
  TextAlign,
} from '../../../helpers/constants/design-system';
import { useI18nContext } from '../../../hooks/useI18nContext';
import { PopoverHeaderProps } from '.';

export const PopoverHeader: React.FC<PopoverHeaderProps> = ({
  children,
  className = '',
  startAccessory,
  endAccessory,
  onClose,
  closeButtonProps,
  onBack,
  backButtonProps,
  ...props
}) => {
  const t = useI18nContext();
  return (
    <HeaderBase
      className={classnames('mm-popover-header', className)}
      startAccessory={
        startAccessory ||
        (onBack && (
          <ButtonIcon
            iconName={IconName.ArrowLeft}
            ariaLabel={t('back')}
            size={ButtonIconSize.Sm}
            onClick={onBack}
            {...backButtonProps}
          />
        ))
      }
      endAccessory={
        endAccessory ||
        (onClose && (
          <ButtonIcon
            iconName={IconName.Close}
            ariaLabel={t('close')}
            size={ButtonIconSize.Sm}
            onClick={onClose}
            {...closeButtonProps}
          />
        ))
      }
      {...props}
    >
      {typeof children === 'string' ? (
        <Text variant={TextVariant.headingSm} textAlign={TextAlign.Center}>
          {children}
        </Text>
      ) : (
        children
      )}
    </HeaderBase>
  );
};
