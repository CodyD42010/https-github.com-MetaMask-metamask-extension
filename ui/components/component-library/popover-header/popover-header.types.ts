import React from 'react';
import type { ButtonIconProps } from '../button-icon/button-icon.types';
import type { HeaderBaseProps } from '../header-base';

export interface PopoverHeaderProps extends HeaderBaseProps {
  /**
   * The contents within the PopoverHeader positioned middle (popular for title use case)
   */
  children?: React.ReactNode;
  /**
   * Additional classNames to be added to the PopoverHeader component
   */
  className?: string;
  /**
   * The onClick handler for the back `ButtonIcon`
   * When passed this will allow for the back `ButtonIcon` to show
   */
  onBack?: () => void;
  /**
   * The props to pass to the back `ButtonIcon`
   */
  backButtonProps?: ButtonIconProps;
  /**
   * The start (left) content area of PopoverHeader
   * Default to have the back `ButtonIcon` when `onBack` is passed, but passing a  `startAccessory` will override this
   */
  startAccessory?: React.ReactNode;
  /**
   * The onClick handler for the close `ButtonIcon`
   * When passed this will allow for the close `ButtonIcon` to show
   */
  onClose?: () => void;
  /**
   * The props to pass to the close `ButtonIcon`
   */
  closeButtonProps?: ButtonIconProps;
  /**
   * The end (right) content area of PopoverHeader
   * Default to have the close `ButtonIcon` when `onClose` is passed, but passing a  `endAccessory` will override this
   */
  endAccessory?: React.ReactNode;
}
