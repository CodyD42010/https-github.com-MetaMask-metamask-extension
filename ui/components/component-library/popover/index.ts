export { Popover } from './popover';
export { PopoverPosition, PopoverRole } from './popover.types';
export type { PopoverProps } from './popover.types';
