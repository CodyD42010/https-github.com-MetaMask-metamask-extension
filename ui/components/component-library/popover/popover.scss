
.mm-popover {
  // TODO: remove when using HeaderBase

  &__header {
    position: relative;

    &-button-back {
      position: absolute;
      top: 0;
      left: 0;
    }

    &-button-close {
      position: absolute;
      top: 0;
      right: 0;
    }
  }

  /* Hide the popper when the reference is hidden */
  &--reference-hidden[data-popper-reference-hidden="true"] {
    visibility: hidden;
    pointer-events: none;

    > .mm-popover__arrow::before {
      visibility: hidden;
    }
  }
}

.mm-popover__arrow,
.mm-popover__arrow::before {
  position: absolute;
  width: 8px;
  height: 8px;
  background: inherit;
}

.mm-popover__arrow {
  width: 40px;
  height: 40px;
  visibility: hidden;
}

.mm-popover__arrow::before {
  display: block;
  background-color: inherit;
  border: 1px solid;
  border-left-color: inherit;
  border-top-color: inherit;
  border-bottom-color: transparent;
  border-right-color: transparent;
  visibility: visible;
  content: '';
  transform: rotate(45deg);
  border-radius: 2px  0 0 0;
}

.mm-popover[data-popper-placement^='top'] > .mm-popover__arrow {
  bottom: -20px;

  &::before {
    transform: rotate(-135deg);
  }
}

.mm-popover[data-popper-placement^='bottom'] > .mm-popover__arrow {
  top: -20px;
}

.mm-popover[data-popper-placement^='left'] > .mm-popover__arrow {
  right: -20px;

  &::before {
    transform: rotate(135deg);
  }
}

.mm-popover[data-popper-placement^='right'] > .mm-popover__arrow {
  left: -20px;

  &::before {
    transform: rotate(-45deg);
  }
}
