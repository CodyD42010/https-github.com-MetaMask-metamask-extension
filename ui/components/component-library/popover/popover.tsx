import React, { useState, useEffect } from 'react';
import { createPortal } from 'react-dom';
import { usePopper } from 'react-popper';
import classnames from 'classnames';
import {
  AlignItems,
  BackgroundColor,
  BorderColor,
  BorderRadius,
  Display,
  JustifyContent,
} from '../../../helpers/constants/design-system';
import Box from '../../ui/box/box';
import { PopoverPosition, PopoverProps, PopoverRole } from '.';

export const Popover = ({
  children,
  className = '',
  position = PopoverPosition.Auto,
  role = PopoverRole.Tooltip,
  hasArrow = false,
  matchWidth,
  preventOverflow = false,
  offset = [0, 8],
  flip = false,
  referenceHidden = true,
  referenceElement,
  isOpen,
  title,
  isPortal = false,
  arrowProps,
  onPressEscKey,
  ...props
}: PopoverProps) => {
  const [popperElement, setPopperElement] = useState<HTMLElement | null>(null);
  const [arrowElement, setArrowElement] = useState<HTMLElement | null>(null);

  // Define Popper options
  const { styles, attributes } = usePopper(referenceElement, popperElement, {
    placement: position,
    modifiers: [
      {
        name: 'preventOverflow',
        enabled: position === PopoverPosition.Auto ? true : preventOverflow,
      },
      {
        name: 'flip',
        enabled: position === PopoverPosition.Auto ? true : flip,
      },
      {
        name: 'arrow',
        enabled: hasArrow,
        options: {
          element: arrowElement,
        },
      },
      {
        name: 'offset',
        options: {
          offset,
        },
      },
    ],
  });

  // Define width to match reference element or auto
  const contentStyle = {
    width: matchWidth ? referenceElement?.clientWidth : 'auto',
  };

  // Esc key press
  const handleEscKey = (event: KeyboardEvent) => {
    if (event.key === 'Escape') {
      // Close the popover when the "Esc" key is pressed
      if (onPressEscKey) {
        onPressEscKey();
      }
    }
  };

  useEffect(() => {
    document.addEventListener('keydown', handleEscKey);

    return () => {
      document.removeEventListener('keydown', handleEscKey);
    };
  }, [onPressEscKey]);

  const PopoverContent = (
    <Box
      borderColor={BorderColor.borderMuted}
      borderRadius={BorderRadius.LG}
      backgroundColor={BackgroundColor.backgroundDefault}
      padding={4}
      role={role}
      className={classnames(
        'mm-popover',
        {
          'mm-popover--open': Boolean(isOpen),
          'mm-popover--reference-hidden': Boolean(referenceHidden),
        },
        className,
      )}
      ref={setPopperElement}
      {...attributes.popper}
      {...props}
      style={{ ...styles.popper, ...contentStyle, ...props.style }}
    >
      {children}
      {hasArrow && (
        <Box
          borderColor={BorderColor.borderMuted}
          className={classnames('mm-popover__arrow')}
          ref={setArrowElement}
          display={Display.Flex}
          justifyContent={JustifyContent.center}
          alignItems={AlignItems.center}
          style={styles.arrow}
          {...attributes.arrow}
          {...arrowProps}
        />
      )}
    </Box>
  );

  return (
    <>
      {isPortal
        ? isOpen && createPortal(PopoverContent, document.body)
        : isOpen && PopoverContent}
    </>
  );
};
