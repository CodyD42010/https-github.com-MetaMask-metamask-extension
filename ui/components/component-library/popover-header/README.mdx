import { Story, Canvas, ArgsTable } from '@storybook/addon-docs';
import { PopoverHeader } from './popover-header';

# PopoverHeader

`PopoverHeader` handles the title, close button and back button for all `Popover` components. It is built on top of the [HeaderBase](/docs/components-componentlibrary-headerbase--default-story) component

<Canvas>
  <Story id="components-componentlibrary-popoverheader--default-story" />
</Canvas>

## Props

The `PopoverHeader` accepts all props below as well as all [Box](/docs/ui-components-ui-box-box-stories-js--default-story#props) component props

<ArgsTable of={PopoverHeader} />

### Children

The title of the `PopoverHeader` component. Passing a `string` will render the content inside of a `Text` component. Passing any other type will render the content as is.

<Canvas>
  <Story id="components-componentlibrary-popoverheader--children" />
</Canvas>

```jsx
import {
  TextVariant,
  TextAlign,
  DISPLAY,
  FLEX_DIRECTION,
  AlignItems,
  JustifyContent,
} from '../../../helpers/constants/design-system';

import { PopoverHeader,  AvatarAccount, Text } from '../../component-library';

<PopoverHeader {...args} marginBottom={4}>
  Children as string
</PopoverHeader>
<PopoverHeader
  {...args}
  childrenWrapperProps={{
    display: DISPLAY.FLEX,
    flexDirection: FLEX_DIRECTION.COLUMN,
    alignItems: AlignItems.center,
    justifyContent: JustifyContent.center,
  }}
>
  <AvatarAccount address="0x1234" />
  <Text variant={TextVariant.headingSm} textAlign={TextAlign.Center}>
    Custom header using multiple components
  </Text>
</PopoverHeader>
```

### onBack

Use the onClick handler `onBack` prop to render the `ButtonIcon` back button in the startAccessory position.

Use the `backButtonProps` prop to pass additional props to the `ButtonIcon` back button.

<Canvas>
  <Story id="components-componentlibrary-popoverheader--on-back" />
</Canvas>

```jsx
import { PopoverHeader } from '../../component-library';

<PopoverHeader onBack={() => console.log('Back button click')}>
  OnBack Demo
</PopoverHeader>;
```

### onClose

Use the onClick handler `onClose` prop to render the `ButtonIcon` back button in the endAccessory position.

Use the `backButtonProps` prop to pass additional props to the `ButtonIcon` back button.

<Canvas>
  <Story id="components-componentlibrary-popoverheader--on-close" />
</Canvas>

```jsx
import { PopoverHeader } from '../../component-library';

<PopoverHeader onClose={() => console.log('Back button click')}>
  OnClose Demo
</PopoverHeader>;
```

### startAccessory

Use the `startAccessory` prop to render a component in the startAccessory position. This will override the default back `ButtonIcon`.

<Canvas>
  <Story id="components-componentlibrary-popoverheader--start-accessory" />
</Canvas>

```jsx
import { PopoverHeader, Button, BUTTON_SIZES } from '../../component-library';

<PopoverHeader startAccessory={<Button size={BUTTON_SIZES.SM}>Demo</Button>}>
  StartAccessory
</PopoverHeader>;
```

### endAccessory

Use the `endAccessory` prop to render a component in the endAccessory position. This will override the default close `ButtonIcon`.

<Canvas>
  <Story id="components-componentlibrary-popoverheader--end-accessory" />
</Canvas>

```jsx
import { PopoverHeader, Button, BUTTON_SIZES } from '../../component-library';

<PopoverHeader endAccessory={<Button size={BUTTON_SIZES.SM}>Demo</Button>}>
  EndAccessory
</PopoverHeader>;
```
