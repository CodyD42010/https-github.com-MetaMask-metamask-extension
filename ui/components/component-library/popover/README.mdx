import { Story, Canvas, ArgsTable } from '@storybook/addon-docs';
import { Popover } from './popover';

# Popover

Popover is an overlay that appears by the trigger used for menus, additional contents, and contains at least one focusable element.

<Canvas>
  <Story id="components-componentlibrary-popover--default-story" />
</Canvas>

## Props

The `Popover` accepts all props below as well as all [Box](/docs/ui-components-ui-box-box-stories-js--default-story#props) component props

<ArgsTable of={Popover} />

### Reference Element

The `referenceElement` prop is required and used to position the popover relative to the reference element.

```jsx
import { useState } from 'react';
import { Popover } from '../../ui/component-library';

const [referenceElement, setReferenceElement] = useState();

const setBoxRef = (ref) => {
setReferenceElement(ref);
};

<Box ref={setBoxRef}></Box>
<Popover referenceElement={referenceElement}>Reference Element</Popover>
```

### Children

Popover accepts any children and has a default padding of `4` (16px).

```jsx
import {
  Popover,
  Text,
  Icon,
  IconSize,
  IconName,
} from '../../ui/component-library';

<Popover>
  <Text>
    Demo of popover with children.
    <Icon size={IconSize.Inherit} name={IconName.Info} />
  </Text>
</Popover>;
```

### Position

Use the `position` prop with the `PopoverPosition` enum to set the position of the popover relative to the reference element.

Default is `PopoverPosition.Auto`

```jsx
import { Popover, PopoverPosition } from '../../ui/component-library';

<Popover position={PopoverPosition.Auto}>Auto</Popover>
<Popover position={PopoverPosition.AutoStart}>AutoStart</Popover>
<Popover position={PopoverPosition.AutoEnd}>AutoEnd</Popover>
<Popover position={PopoverPosition.Top}>Top</Popover>
<Popover position={PopoverPosition.TopStart}>TopStart</Popover>
<Popover position={PopoverPosition.TopEnd}>TopEnd</Popover>
<Popover position={PopoverPosition.Right}>Right</Popover>
<Popover position={PopoverPosition.RightStart}>RightStart</Popover>
<Popover position={PopoverPosition.RightEnd}>RightEnd</Popover>
<Popover position={PopoverPosition.Bottom}>Bottom</Popover>
<Popover position={PopoverPosition.BottomStart}>BottomStart</Popover>
<Popover position={PopoverPosition.BottomEnd}>BottomEnd</Popover>
<Popover position={PopoverPosition.Left}>Left</Popover>
<Popover position={PopoverPosition.LeftStart}>LeftStart</Popover>
<Popover position={PopoverPosition.LeftEnd}>LeftEnd</Popover>
```

### Is Portal

The `isPortal` prop is a boolean that when set to true, causes the Popover to be rendered as a separate DOM element at the end of the document body.
Default `true`

```jsx
import { Popover } from '../../ui/component-library';

<Popover hasArrow>Popover with arrow</Popover>;
```

### Has Arrow

Use the `hasArrow` boolean to add an arrow to the popover.

```jsx
import { Popover } from '../../ui/component-library';

<Popover hasArrow>Popover with arrow</Popover>;
```

### Is Open

Use the `isOpen` boolean to control the visibility of the popover.

```jsx
import { Popover } from '../../ui/component-library';

<Popover isOpen={true}>Popover with arrow</Popover>;
```

### Flip

Use the `flip` boolean to flip the popover to the opposite side of the reference element if there is not enough space.
For `PopoverPosition.Auto` this will become true.

```jsx
import { Popover } from '../../ui/component-library';

<Popover flip={true}>Flip demo</Popover>;
```

### Prevent Overflow

Use the `preventOverflow` boolean to prevent the popover from overflowing the viewport.
For `PopoverPosition.Auto` this will become true.

```jsx
import { Popover } from '../../ui/component-library';

<Popover preventOverflow={true}>Prevent overflow demo</Popover>;
```

### Reference Hidden

Use the `referenceHidden` boolean to hide the Popover when the reference element is no longer visible in the viewport.

```jsx
import { Popover } from '../../ui/component-library';

<Popover referenceHidden={true}>Reference hidden demo</Popover>;
```

### Match Width

Use the `matchWidth` boolean to match the width of the popover to the reference element.

```jsx
import { Popover } from '../../ui/component-library';

<Popover matchWidth={true}>Match width demo</Popover>;
```

### Role

Use the `role` prop with `PopoverRole` enum to set the role of the popover.
`PopoverRole.Dialog` if the content is interactive, or `PopoverRole.Tooltip` for purely informational popovers.

Default: `PopoverRole.Tooltip`

```jsx
import { Popover, PopoverRole } from '../../ui/component-library';

<Popover role={PopoverRole.Tooltip}>PopoverRole.Tooltip</Popover>;
<Popover role={PopoverRole.Dialog}>PopoverRole.Dialog</Popover>;
```

### Offset

Use the `offset` prop to pass an array of two numbers to offset the popover from the reference element.
Default is `[0, 8]`
First number controls the skidding offset and the second number controls the distance offset.

```jsx
import { Popover } from '../../ui/component-library';

<Popover offset={[0, 32]}>offset override to [0,32]</Popover>;
```

### Esc Key Close

TODO: Add description

```jsx
import { Popover } from '../../ui/component-library';

<Popover>ToDO:</Popover>;
```

### With PopoverHeader

TODO: Add description

```jsx
import { Popover } from '../../ui/component-library';

<Popover>ToDO:</Popover>;
```

### Mouse Event Demo

TODO: Add description

```jsx
import { Popover } from '../../ui/component-library';

<Popover>ToDO:</Popover>;
```

### On Focus/Blur Demo

TODO: Add description

```jsx
import { Popover } from '../../ui/component-library';

<Popover>ToDO:</Popover>;
```
