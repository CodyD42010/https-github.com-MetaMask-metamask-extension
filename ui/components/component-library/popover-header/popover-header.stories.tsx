import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { BUTTON_SIZES, Button } from '..';
import { PopoverHeader } from './popover-header';
import README from './README.mdx';

export default {
  title: 'Components/ComponentLibrary/PopoverHeader',
  component: PopoverHeader,
  parameters: {
    docs: {
      page: README,
    },
  },
  argTypes: {
    children: { control: 'text' },
    className: { control: 'text' },
    onBack: { action: 'onBack' },
    onClose: { action: 'onClose' },
  },
  args: {
    children: 'PopoverHeader',
  },
} as ComponentMeta<typeof PopoverHeader>;

const Template: ComponentStory<typeof PopoverHeader> = (args) => {
  return <PopoverHeader {...args}>PopoverHeader</PopoverHeader>;
};

export const DefaultStory = Template.bind({});
DefaultStory.storyName = 'Default';

export const Children: ComponentStory<typeof PopoverHeader> = (args) => (
  <PopoverHeader {...args} />
);

Children.args = {
  children: 'PopoverHeader Title',
};

export const OnBack: ComponentStory<typeof PopoverHeader> = (args) => (
  <PopoverHeader {...args}>OnBack Demo</PopoverHeader>
);

export const OnClose: ComponentStory<typeof PopoverHeader> = (args) => (
  <PopoverHeader {...args}>OnClose Demo</PopoverHeader>
);

export const StartAccessory: ComponentStory<typeof PopoverHeader> = (args) => (
  <PopoverHeader
    startAccessory={<Button size={BUTTON_SIZES.SM}>Demo</Button>}
    {...args}
  >
    StartAccessory
  </PopoverHeader>
);

export const EndAccessory: ComponentStory<typeof PopoverHeader> = (args) => (
  <PopoverHeader
    endAccessory={<Button size={BUTTON_SIZES.SM}>Demo</Button>}
    {...args}
  >
    EndAccessory
  </PopoverHeader>
);
