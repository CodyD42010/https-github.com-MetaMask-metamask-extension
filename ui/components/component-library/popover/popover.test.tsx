/* eslint-disable jest/require-top-level-describe */
import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import { PopoverHeader } from '../popover-header';
import { Popover } from './popover';
import { PopoverPosition } from './popover.types';

describe('Popover', () => {
  it('should render popover element correctly', () => {
    const { getByTestId, getByText, container } = render(
      <Popover data-testid="popover" isOpen={true} isPortal={false}>
        Popover
      </Popover>,
    );
    expect(getByText('Popover')).toBeDefined();
    expect(container.querySelector('popover')).toBeDefined();
    expect(getByTestId('popover')).toHaveClass('mm-popover');
    expect(container).toMatchSnapshot();
  });

  it('should render popover children', () => {
    const { getByText } = render(
      <Popover isOpen={true}>Popover content goes here</Popover>,
    );
    expect(getByText('Popover content goes here')).toBeDefined();
  });

  it('should render popover back button', () => {
    const onBackTest = jest.fn();
    const { getByTestId } = render(
      <Popover data-testid="popover" isOpen={true}>
        <PopoverHeader
          onBack={onBackTest}
          backButtonProps={{ 'data-testid': 'back' }}
        >
          onClose Test
        </PopoverHeader>
      </Popover>,
    );

    const backButton = getByTestId('back');
    fireEvent.click(backButton);

    expect(onBackTest).toHaveBeenCalled();
  });

  it('should render popover close button', () => {
    const onCloseTest = jest.fn();
    const { getByTestId } = render(
      <Popover data-testid="popover" isOpen={true}>
        <PopoverHeader
          onClose={onCloseTest}
          closeButtonProps={{ 'data-testid': 'close' }}
        >
          onClose Test
        </PopoverHeader>
      </Popover>,
    );

    const closeButton = getByTestId('close');
    fireEvent.click(closeButton);

    expect(onCloseTest).toHaveBeenCalled();
  });

  it('should render popover position', () => {
    const { getByText } = render(
      <>
        <Popover
          isOpen={true}
          data-testid={PopoverPosition.Auto}
          position={PopoverPosition.Auto}
        >
          {PopoverPosition.Auto}
        </Popover>

        <Popover
          isOpen={true}
          data-testid={PopoverPosition.Top}
          position={PopoverPosition.Top}
        >
          {PopoverPosition.Top}
        </Popover>

        <Popover
          isOpen={true}
          data-testid={PopoverPosition.TopStart}
          position={PopoverPosition.TopStart}
        >
          {PopoverPosition.TopStart}
        </Popover>

        <Popover
          isOpen={true}
          data-testid={PopoverPosition.TopEnd}
          position={PopoverPosition.TopEnd}
        >
          {PopoverPosition.TopEnd}
        </Popover>

        <Popover
          isOpen={true}
          data-testid={PopoverPosition.Right}
          position={PopoverPosition.Right}
        >
          {PopoverPosition.Right}
        </Popover>

        <Popover
          isOpen={true}
          data-testid={PopoverPosition.RightStart}
          position={PopoverPosition.RightStart}
        >
          {PopoverPosition.RightStart}
        </Popover>

        <Popover
          isOpen={true}
          data-testid={PopoverPosition.RightEnd}
          position={PopoverPosition.RightEnd}
        >
          {PopoverPosition.RightEnd}
        </Popover>

        <Popover
          isOpen={true}
          data-testid={PopoverPosition.Bottom}
          position={PopoverPosition.Bottom}
        >
          {PopoverPosition.Bottom}
        </Popover>

        <Popover
          isOpen={true}
          data-testid={PopoverPosition.BottomStart}
          position={PopoverPosition.BottomStart}
        >
          {PopoverPosition.BottomStart}
        </Popover>

        <Popover
          isOpen={true}
          data-testid={PopoverPosition.BottomEnd}
          position={PopoverPosition.BottomEnd}
        >
          {PopoverPosition.BottomEnd}
        </Popover>

        <Popover
          isOpen={true}
          data-testid={PopoverPosition.Left}
          position={PopoverPosition.Left}
        >
          {PopoverPosition.Left}
        </Popover>

        <Popover
          isOpen={true}
          data-testid={PopoverPosition.LeftStart}
          position={PopoverPosition.LeftStart}
        >
          {PopoverPosition.LeftStart}
        </Popover>

        <Popover
          isOpen={true}
          data-testid={PopoverPosition.LeftEnd}
          position={PopoverPosition.LeftEnd}
        >
          {PopoverPosition.LeftEnd}
        </Popover>
      </>,
    );

    expect(getByText(PopoverPosition.Auto)).toBeDefined();
    expect(getByText(PopoverPosition.Top)).toBeDefined();
    expect(getByText(PopoverPosition.TopStart)).toBeDefined();
    expect(getByText(PopoverPosition.TopEnd)).toBeDefined();
    expect(getByText(PopoverPosition.Right)).toBeDefined();
    expect(getByText(PopoverPosition.RightStart)).toBeDefined();
    expect(getByText(PopoverPosition.RightEnd)).toBeDefined();
    expect(getByText(PopoverPosition.Bottom)).toBeDefined();
    expect(getByText(PopoverPosition.BottomStart)).toBeDefined();
    expect(getByText(PopoverPosition.BottomEnd)).toBeDefined();
    expect(getByText(PopoverPosition.Left)).toBeDefined();
    expect(getByText(PopoverPosition.LeftStart)).toBeDefined();
    expect(getByText(PopoverPosition.LeftEnd)).toBeDefined();
  });

  it('should render an arrow on popover', () => {
    const { getByTestId } = render(
      <Popover data-testid="popover" isOpen={true} hasArrow>
        Popover
      </Popover>,
    );

    const arrowElement =
      getByTestId('popover').querySelector('.mm-popover__arrow');
    expect(arrowElement).toHaveClass('mm-popover__arrow');
  });
});
