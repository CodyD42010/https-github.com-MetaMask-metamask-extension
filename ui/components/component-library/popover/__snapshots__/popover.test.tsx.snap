// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover should render popover element correctly 1`] = `
<div>
  <div
    class="box mm-popover mm-popover--open mm-popover--reference-hidden box--padding-4 box--flex-direction-row box--background-color-background-default box--rounded-lg box--border-color-border-muted box--border-style-solid box--border-width-1"
    data-testid="popover"
    role="tooltip"
    style="position: absolute; left: 0px; top: 0px; width: auto;"
  >
    Popover
  </div>
</div>
`;
